public/app/plugins/datasource/loki/configuration/DerivedField.tsx

```diff
@@ -60,101 +65,108 @@ export const DerivedField = (props: Props) => {
+        >
+          <Input value={value.matcherRegex} onChange={handleChange('matcherRegex')} />
+        </Field>
+        <Field label="">
+          <Button
+            variant="destructive"
+            title="Remove field"
+            icon="times"
+            onClick={(event) => {
+              event.preventDefault();
+              onDelete();
+            }}
+          />
+        </Field>
       </div>
 
       <div className="gf-form">
-        <FormField
-          labelWidth={10}
-          label={showInternalLink ? 'Query' : 'URL'}
-          inputEl={
-            <DataLinkInput
-              placeholder={showInternalLink ? '${__value.raw}' : 'http://example.com/${__value.raw}'}
-              value={value.url || ''}
-              onChange={(newValue) =>
-                onChange({
-                  ...value,
-                  url: newValue,
-                })
-              }
-              suggestions={suggestions}
+        <Field label={showInternalLink ? 'Query' : 'URL'} className={styles.urlField}>
+          <DataLinkInput
+            placeholder={showInternalLink ? '${__value.raw}' : 'http://example.com/${__value.raw}'}
+            value={value.url || ''}
+            onChange={(newValue) =>
+              onChange({
+                ...value,
+                url: newValue,
+              })
+            }
+            suggestions={suggestions}
+          />
+        </Field>
+        <Field
+          className={styles.urlDisplayLabelField}
+          label={
+            <TooltipLabel
+              label="URL Label"
+              content="Use to override the button label when this derived field is found in a log."
             />
           }
-          className={styles.urlField}
-        />
-        <FormField
-          className={styles.urlDisplayLabelField}
-          labelWidth={10}
-          inputWidth={null}
-          label="URL Label"
-          type="text"
-          value={value.urlDisplayLabel}
-          onChange={handleChange('urlDisplayLabel')}
-          tooltip={'Use to override the button label when this derived field is found in a log.'}
-        />
+        >
+          <Input value={value.urlDisplayLabel} onChange={handleChange('urlDisplayLabel')} />
+        </Field>
       </div>
 
-      <div className={styles.row}>
-        <Switch
-          label="Internal link"
-          checked={showInternalLink}
-          onChange={() => {
-            if (showInternalLink) {
-              onChange({
-                ...value,
-                datasourceUid: undefined,
-              });
-            }
-            setShowInternalLink(!showInternalLink);
-          }}
-        />
+      <div className="gf-form">
+        <Field label="Internal link" className={styles.internalLink}>
+          <Switch
+            value={showInternalLink}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+              const checked = e.currentTarget.checked;
```

Comment:
nitpick: if you want to you can write it this way too, but it's as you prefer.

```suggestion
              const { checked } = e.currentTarget;
```